### ISSUE TITLE
Implement Kohyama model (New (simple) strategy to test)

### ISSUE THREAD

**Issue by dfalster** [traitecoevo/plant#218]
here's an idea @rafaqz --What about implementing the model from http://doi.org/10.2307/2261230: Kohyama 1993: "Size-structured tree populations in gap-dynamic forest: the forest architecture hypothesis for the stable coexistence of species". Also interesting scientifically to see whether their results holds up.

It's a very simple model (for trees), so a good test for the current work enabling alternative strategies and competitive environments?

**Comment by dfalster** [comment]
It contains some similar features but the entire model is much simpler. It's an empirically motivated model with only 1 equation for each of growth, mortality, reproduction:
<img width="416" alt="image" src="https://user-images.githubusercontent.com/3589590/68097228-f44d3580-ff09-11e9-8870-1bbf17c04c68.png">

As such, there are no trait-based trade-offs.But one can connect different paremeters from the above to induce a trade-off. 

Competition is based on total basal area:

<img width="269" alt="image" src="https://user-images.githubusercontent.com/3589590/68097279-45f5c000-ff0a-11e9-9d94-496fd25323e8.png">,
(summing across all species), where f is the density of plants per unit ground area per unit size at given stand age (n in our model).

**Comment by rafaqz** [comment]
Ok that makes sense. I'll have a look at implementing it tomorrow. 

Do you think this can be done by simply adding another strategy, or can you forsee other required changes?

**Comment by dfalster** [comment]
Should be mostly straight forward. But completion will be handled differently, as there is only basal area, not leaf area, and a different calculation for effect of competition on growth. But that is why it is a good challenge: will help us sort out generalizing of environment-competition part.

**Comment by dfalster** [comment]
@aornugent made a good start on this, but we encountered some issues (see discussion under #247)

I suggested holding off investigating, until we resolved some issue with the main class (see #244, #241 #243)
 
I've now pushed the Kohyama model onto a new branch, to make use of the updated material.

But alas, the original error described by @aornugent in #247 persists. 

After a couple of iterations it runs into the following error:
```
Error in SCM___K93__K93_Env__run_next(self) : Interpolated function as refined as currently possible 
```
_Originally posted by @aornugent in https://github.com/traitecoevo/plant/pull/247#issuecomment-664212711_


I'm now wondering if it's because of the step function in the competition function. 

```
double K93_Strategy::compute_competition(double z, double size) const {
  return (z > size)?0:size_to_basal_area(size);
 };
```

The adaptive interpolator will struggle to resolve this. 

Next suggest trying a continuous function.
